apps/lib/log.c

```diff
@@ -56,18 +56,31 @@ static void log_with_prefix(const char *prog, const char *fmt, va_list ap)
     BIO_free(pre);
 }
 
+/*
+ * Unfortuantely, C before C99 does not define va_copy, so we must
+ * check if it can be assume to be present.  We do that with an internal
+ * antifeature macro.
+ * C versions since C94 define __STDC_VERSION__, so it's enough to
+ * check its existence and value.
+ */
+#undef OSSL_NO_C99
+#if !defined(__STDC_VERSION__) || __STDC_VERSION__ + 0 < 199900L
```

Comment:
https://github.com/ocornut/imgui/issues/1085 does this as
#if !defined(__STRICT_ANSI__) || __STDC_VERSION__ + 0 >= 199900L \

Reply:
Yup.  That works for GCC and clang.
Maybe read this PR's description once more?

Reply:
... oh, and the comment?

Reply:
I did.. My point is that it only checked #ifdef __STRICT_ANSI__  before..

Reply:
Okie.  That doesn't tell me that something's wrong with the test I have, so I guess I can take that as a side note then?

Reply:
Surely you can just do

```suggestion
#if __STDC_VERSION__ < 199900L
```

since, per the C standard, anything not defined in a pre-processor line evaluates to zero:

> After all replacements due to macro expansion and the defined unary
> operator have been performed, all remaining identifiers (including those lexically
> identical to keywords) are replaced with the pp-number 0,

Reply:
@tom-cosgrove-arm, how far back is that applicable?  I forget.

Reply:
Oh, I just tried that using a `--strict-warnings` configuration with `-ansi` added:

```
../master/apps/lib/log.c: At top level:
../master/apps/lib/log.c:67:5: error: "__STDC_VERSION__" is not defined, evaluates to 0 [-Werror=undef]
   67 | #if __STDC_VERSION__ < 199900L
      |     ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

I'm afraid that won't do.